testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -187,6 +187,162 @@ public void testReadSparkCreatedTable(String sparkTableFormat, String expectedTr
+                        "TIMESTAMP '1965-09-10 13:00:00.123456', " + // a_timestamp_micros
+                        "TIMESTAMP '1965-09-10 13:00:00.123456789', " + // a_timestamp_nanos
+                        "'dummy')");
+
+        List<Row> sparkExpected = List.of(
+                row(nCopies(16, null).toArray()),
+                row(
+                        true, // a_boolean
+                        (byte) 127, // a_tinyint
+                        (short) 32767, // a_smallint
+                        1000000000, // an_integer
+                        1000000000000000L, // a_bigint
+                        10000000.123F, // a_real
+                        100000000000.123, // a_double
+                        new BigDecimal("1234567.8901"), // a_short_decimal
+                        new BigDecimal("1234567890123456789.0123456"), // a_long_decimal
+                        "some string", // a_string
+                        java.sql.Date.valueOf(LocalDate.of(2005, 9, 10)), // a_date
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0)), // a_timestamp_seconds
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0, 123_000_000)), // a_timestamp_millis
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_micros
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_nanos; note that Spark timestamp has microsecond precision
+                        "dummy"),
+                row(
+                        false, // a_bigint
+                        (byte) -128, // a_tinyint
+                        (short) -32768, // a_smallint
+                        -1000000012, // an_integer
+                        -1000000000000012L, // a_bigint
+                        -10000000.123F, // a_real
+                        -100000000000.123, // a_double
+                        new BigDecimal("-1234567.8901"), // a_short_decimal
+                        new BigDecimal("-1234567890123456789.0123456"), // a_long_decimal
+                        "", // a_string
+                        java.sql.Date.valueOf(LocalDate.of(1965, 9, 10)), // a_date
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0)), // a_timestamp_seconds
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0, 123_000_000)), // a_timestamp_millis
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_micros
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(1965, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_nanos; note that Spark timestamp has microsecond precision
+                        "dummy"));
+        List<Row> trinoExpected = List.of(
```

Comment:
Having separate trinoExpected, sparkExpected is error-prone.
The expected results should be the same in both engines.
What is the difference?

Reply:
The difference is that Spark does not support nanoseconds.

Reply:
Let's have "all types" test where with millisecond precision (default), where expected results are same for Spark and Trino.